pages/client-libraries/java.mdx

```diff
@@ -4,105 +4,687 @@ description: Peruse our guide on implementing Java in Memgraph. Comprehensive do
+Also mange transaction have a retry logic, which means that if the transaction fails, it will be retried. Managed transactions are the most flexible way to run a Cypher query.
+
+##### Implicit transactions 
+
+Implicit auto commit queries are the simplest way to run a Cypher query since they won't be automatically retried as with writeTransaction and readTransaction. 
+With implicit auto commit transactions, you don't have the same control of transaction as with manged explicit transactions. 
+
+Here is the example of implicit transaction:
+```java
+public void createNode() {
+    try (Session session = driver.session(SessionConfig.forDatabase("memgraph"))) {
+        session.run("CREATE (n:Technology {name:'Memgraph'})");
+    }
+}
+```
+
+As you can see in this case you can run a simple query without any transaction management. This simplifys the experience but you lose some flexibility when running the queries. 
+Since explicit transaction are multi-statement transactions, creating an index, constraints, auth config queries are not possible in multi-line transactions. Hence, you need to use implicit transactions for those queries.
+
+#### Asynchronous transaction managment
+
+##### Session management
+
+Asynchronous sessions provide a API to work with futures. This allows client applications to work within asynchronous code bases. Asynchronous sessions are not thread safe, so you should create a new session for each thread.
+Session lifetime begins with session construction, session is typicaly closed automatically when all results are consumed. 
+
+The session is opened with the following code snippet: 
+
+```java
+ var session = driver.session(AsyncSession.class);
+```
+
+Via async session API you can use following transaction management methods:
+
+ - `writeTransactionAsync` - executes a manged transaction workload in write access mode with retry logic.
+ - `readTransactionAsync` - executes a managed transaction workload in read access mode with retry logic.
+ - `runAsync` - executes a implicit auto-commit transaction and returns result.
+
+
+##### Managed transactions 
+
+Let's take a look into the basic example if you want to read data from the database via Async session API:
+
+```java
+public CompletionStage<ResultSummary> readNodesAsync() {
+    String query = "MATCH (n:Technology {name: 'Memgraph'}) RETURN n;";
+    var session = driver.session(AsyncSession.class);
+
+    return session.executeReadAsync(tx -> tx.runAsync(query)
+            .thenCompose(cursor -> cursor.forEachAsync(record ->
+                    // asynchronously print every record
+                    System.out.println(record.get(0).asString()))));
+
+}
+```
+
+
+##### Implicit auto-commit asynchronous transactions
+
+Implicit auto commit queries are the simplest way to run a Cypher query since they won't be automatically retried as with writeTransactionAsync and readTransactionAsync.
```

Comment:
```suggestion
Implicit auto-commit queries are the simplest way to run a Cypher query since they aren't automatically retried like `writeTransactionAsync` and `readTransactionAsync` are.
```